### ISSUE TITLE
old references to namelist_definition_clm4_5.xml

### ISSUE THREAD

**Issue by rgknox** [E3SM-Project/E3SM#4130]
I'm working on updating surface files for elm, and ran across some deprecated paths in the scripting.  

For instance,   components/elm/tools/clm4_5/mksurfdata_map/mksurfdata.pl --help was looking for: 

./../../../bld/namelist_files/namelist_definition_clm4_5.xml and generated one of those "root_element" errors, which seems to mean it could not find the xml file, and therefore could not find the root of it.

I did a grep on the code for where we still reference that old file and there are a few places:

``` 
> grep -Ri namelist_definition_clm4_5.xml 
cime/config/e3sm/config_files.xml:      <value component="clm"      >$SRCROOT/components/clm/bld/namelist_files/namelist_definition_clm4_5.xml</value>
components/elm/tools/clm4_5/mksurfdata_map/mksurfdata.pl:my $nldef_file     = "$scrdir/../../../bld/namelist_files/namelist_definition_clm4_5.xml";
components/elm/tools/clm4_5/mksurfdata_map/README.developers:- Add your new file in bld/namelist_files/namelist_definition_clm4_5.xml;
components/elm/tools/clm4_5/mksurfdata_map/README.developers:  mksrf_filename in bld/namelist_files/namelist_definition_clm4_5.xml
components/elm/tools/clm4_5/mksurfdata_map/README.developers:  hgrid, res and lmask in bld/namelist_files/namelist_definition_clm4_5.xml
components/elm/src/main/findHistFields.pl:my $nl_definition_file = "$cfgdir/../../../bld/namelist_files/namelist_definition_clm4_5.xml";
components/elm/src/external_models/fates/main/FatesInterfaceTypesMod.F90:                                         ! See namelist_definition_clm4_5.xml
components/homme/utils/cime/config/e3sm/config_files.xml:      <value component="clm"      >$SRCROOT/components/clm/bld/namelist_files/namelist_definition_clm4_5.xml</value>
```
I'd be happy to update these with the newer file name.  I don't really know this code well or really understand what is going on, or have sense of it systemically, so any guidance is appreciated.

**Comment by rgknox** [comment]
Looks like the queryDefaultNamelist.pl script needs updating too:
```
diff --git a/components/elm/bld/queryDefaultNamelist.pl b/components/elm/bld/queryDefaultNamelist.pl
index b57911b26..848bdd48e 100755
--- a/components/elm/bld/queryDefaultNamelist.pl
+++ b/components/elm/bld/queryDefaultNamelist.pl
@@ -180,9 +180,9 @@ EOF
   my %inputopts;
   my $model                  = $opts{'model'};
   my @nl_definition_files    = ( "$cfgdir/namelist_files/namelist_definition_drv.xml",
-                                 "$cfgdir/namelist_files/namelist_definition_$model.xml" 
+                                 "$cfgdir/namelist_files/namelist_definition.xml" 
                                );
-  $inputopts{empty_cfg_file} = "$cfgdir/config_files/config_definition_$model.xml";
+  $inputopts{empty_cfg_file} = "$cfgdir/config_files/config_definition.xml";
   $inputopts{nldef_files}    = \@nl_definition_files;
   $inputopts{namelist}       = $opts{namelist};
   $inputopts{printing}       = $printing;
@@ -246,7 +246,7 @@ EOF
      $settings{'notest'}       = ! $opts{'test'};
      $settings{'csmdata'}      = $inputopts{csmdata};
   } else {
-     my @files = ( "$cfgdir/namelist_files/namelist_defaults_${model}.xml", 
+     my @files = ( "$cfgdir/namelist_files/namelist_defaults.xml", 
                    "$cfgdir/namelist_files/namelist_defaults_${model}_tools.xml", 
                    "$cfgdir/namelist_files/namelist_defaults_drv.xml",
                    "$cfgdir/namelist_files/namelist_defaults_drydep.xml",
diff --git a/components/elm/src/external_models/fates b/components/elm/src/external_models/fates
index 5534a9405..5701b29d1 160000
--- a/components/elm/src/external_models/fates
+++ b/components/elm/src/external_models/fates
```

**Comment by acme-y9s** [comment]
@rgknox  I checkout the current master last week and met the same issue as yours. But I fixed the problem by modifying the file components/elm/bld/queryDefaultNamelist.pl and mksurfdata.pl to make sure they point to the namelist_defaults.xml, not to namelist_defaults_clm4_5.xml. I can successfully create the name list file for resolution ne120np4, but for some resolution the model still fails to generate name list file.  Just like @aldivi mentioned on E3SM conference page "this step will fail if the required mapping files are not in …/inputdata/lnd/clm2/mappingdata/maps/<res>, or if the definitions of these mapping files are incorrect in namelist_defaults.xml (search for the “map” tag and the “to_hgrid” field that matches your desired resolution, e.g., to_hgrid=”ne30np4”). You should also check in namelist_defaults_clm4_5_tools.xml to make sure that there is a “scripgriddata” tag entry with an “hgrid” field that matches the desired resolution."  Maybe I will submit a pull request to fix the issue.

**Comment by aldivi** [comment]
@acme-y9s I don't think that the missing resolution defaults in these namelist definition files are necessarily "issues." The usual reason why they are not there is because the files do not exist and first have to be created following the instructions here: https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/17498263/CLM4.5+dataset+for+year+1850+at+ne120np4+resolution
There may be some instances where the mapping files do exist are not in the defaults files.

**Comment by acme-y9s** [comment]
@aldivi I meant submit a pull request to fix the issue to point to the right namelist_default.xml, not to namelist_default_clm4_5.xml.  Definitely, for the surface data generation, the actual first step is to create the scripgriddata grid file the given resolution, and then create the mapping files by using the scripgriddata file. And we need to make sure the scripgriddata file for the given resolution in namelist_defaults_clm4_5_tools.xm is existing at the ~/inputdata/lnd/clm2/mappingdata/grids and they are matching in these two places. we also need to make sure the mapping files for the given resolution in namelist_default.xml are existing at ~/inputdata/lnd/clm2/mappingdata/maps, and all the files names are identical for these two places. Otherwise, the mksurfdata.pl will fail to create the namelist file for the next step of make surface data. I agree with for some resolution the mapping files does exist (like ne11np4), but are not in the namelist_default.xml. But it is easy to add to the default file if we want.

**Comment by rljacob** [comment]
Yes please make a PR to fix the ref to namelist_default_clm4_5.xml.  This was probably missed during the clm -> elm rename.

In general, there are many combos of resolution and compset that are not supported.

**Comment by jjbenedict** [comment]
@acme-y9s @aldivi @rgknox @rljacob, I just cloned E3SM-master to cori yesterday I'm still running into this issue when trying to run #E3SMROOT/components/elm/tools/clm4_5/mksurfdata_map/mksurfdata.pl, as Ryan saw in February.  Is someone working on this fix?  Thanks!

**Comment by rljacob** [comment]
Looks like the PR that fixes this, #4197, was just merged to next today.
